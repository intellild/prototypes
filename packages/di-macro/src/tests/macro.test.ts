import plugin from 'babel-plugin-macros';
import pluginTester from 'babel-plugin-tester';
import path from 'path';

pluginTester({
  plugin,
  pluginName: 'injectable-macro',
  babelOptions: {
    filename: __filename,
    presets: ['@babel/preset-typescript'],
    plugins: [
      [
        '@babel/plugin-syntax-decorators',
        {
          legacy: true,
        },
      ],
    ],
  },
  snapshot: true,
  tests: [
    {
      title: 'injectable-macro',
      fixture: path.resolve(__dirname, './fixture1.ts'),
    },
  ],
});
