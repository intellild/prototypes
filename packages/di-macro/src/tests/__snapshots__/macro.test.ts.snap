// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injectable-macro injectable-macro: injectable-macro 1`] = `

import injectable from './lib.proxy.macro';

@injectable()
class B {}

@injectable()
class A {
  constructor(b: B) {}
}

      ↓ ↓ ↓ ↓ ↓ ↓

class B {
  static ɵfac(container, context) {
    return new B();
  }
}

class A {
  constructor(b) {}

  static ɵfac(container, context) {
    return new A(container.resolve(B, context));
  }
}


`;
